### DISCUSSION TITLE
Restoring windows from disk

### DISCUSSION THREAD

**Discussion by iulikolcsar** [kangyu-california/PersistentWindows#158] [General]
I have a problem with the "restore windows from disk" function and I would like to suggest some potential improvements.

I must mention that I use this app a lot, literally every day, and it does its job, it works like a charm, but I noticed that the "restore windows from disk" function, won't work properly if I turn on my monitors when my PC is already ON.

**Example:** I turn ON my PC, while it powers on, I grab some coffee and when I come back, I realize that I forgot to turn on my 2 monitors. I proceed to turn on the monitors, but when they start, all my windows are stacked on the primary monitor instead of taking their place on all of my monitors. 

To solve this, I have to right-click the PersistentWindow tray icon, and click "restore windows from disk". 

I could do this manually every time, but I think automating this could be a nice improvement for everyone who uses this app. 

Could you somehow make the app check if the windows I "captured" to disk are actually in their right place? 
Or perhaps, could you implement a timer? So I could set a time for the windows to restore, let's say, 2 minutes after my PC has started.
That way, even if the windows aren't there at first, the timer would trigger the "restore windows from disk" function 2 minutes later, and eventually, the windows would be in their right place. 

This would help me a lot and even though it seems like a very small issue, it's really important to me and It could give me a better workflow. 

Thank you so much for your time and thank you for this great app!

**Disclaimer:** I hope I didn't offend anyone with this, I am not a programmer and I do not know how to code so I don't know how much I ask of you.

**Comment by kangyu-california** [comment]
Good suggestion, will work on it.

**Reply by iulikolcsar** [reply]
Thank you so much, I highly appreciate this!!

**Comment by AuricMists** [comment]
Hi, this functionality seemed to be in place a couple of months ago but disappeared during (I think) the last update. I have to manually "Restore windows from disk" again.  Will it be going back in?

Thanks for maintaining this invaluable tool.

**Comment by kangyu-california** [comment]
@AuricMists 

I get the enhancement requirement is 
1. Automatically restore existing window to its last captured position when PW starts.
2. Automatically restore newly launched window to its last captured position.

Will work on it.

**Comment by AuricMists** [comment]
Appreciate it.

Not too much of an issue for me but I was curious as to why the
functionality changed.
